prov/util/src/rxm_av.c

```diff
@@ -98,23 +98,32 @@ static void rxm_free_peer(struct util_peer_addr *peer)
 	ofi_ibuf_free(peer);
 }
 
-struct util_peer_addr *
-util_get_peer(struct rxm_av *av, const void *addr)
+
+int util_get_peer(struct rxm_av *av, const void *addr, struct util_peer_addr **peer,
+		  uint64_t flags)
 {
-	struct util_peer_addr *peer;
 	struct ofi_rbnode *node;
+	int ret;
 
 	ofi_mutex_lock(&av->util_av.lock);
 	node = ofi_rbmap_find(&av->addr_map, (void *) addr);
 	if (node) {
-		peer = node->data;
-		peer->refcnt++;
+		*peer = node->data;
+		(*peer)->refcnt++;
 	} else {
-		peer = rxm_alloc_peer(av, addr);
-	}
+		if (flags & FI_NO_CONNECT) {
+			*peer = NULL;
+			ret = -FI_EHOSTUNREACH;
+			goto unlock;
+		}
 
+		*peer = rxm_alloc_peer(av, addr);
+		if (!*peer)
+			ret = -FI_ENOMEM;
+	}
+unlock:
```

Comment:
nit: This can be restructured as:
```
if (node) {
} else if (flags & ...) {
} else {
}
```

Reply:
I would switch to if-else if-else.  This removes the goto and clearly shows the order of the checks.